src/js/actions/deviceActions.js

```diff
@@ -1061,3 +1059,74 @@ export const setDeviceTwin = (deviceId, integration, settings) => (dispatch, get
         })
       );
     });
+
+const prepareSearchArguments = ({ filters, group, state, status }) => {
+  const { filterTerms } = convertDeviceListStateToFilters({ filters, group, offlineThreshold: state.app.offlineThreshold, selectedIssues: [], status });
+  const { columnSelection = [] } = getUserSettings(state);
+  const selectedAttributes = columnSelection.map(column => ({ attribute: column.key, scope: column.scope }));
+  const attributes = [...defaultAttributes, { scope: 'identity', attribute: getIdAttribute(state).attribute }, ...selectedAttributes];
+  return { attributes, filterTerms };
+};
+
+export const getSystemDevices =
+  (id, options = {}) =>
+  (dispatch, getState) => {
+    const { page = defaultPage, perPage = defaultPerPage } = options;
+    const state = getState();
+    const { hasFullFiltering } = getTenantCapabilities(state);
+    if (!hasFullFiltering) {
+      return Promise.resolve();
+    }
+    const filters = [{ ...emptyFilter, key: 'id', operator: DEVICE_FILTERING_OPTIONS.$in.key, value: id }];
+    const { attributes, filterTerms } = prepareSearchArguments({ filters, state, status: 'any' });
+
+    return GeneralApi.post(getSearchEndpoint(state.app.features.hasReporting), {
+      page,
+      per_page: perPage,
+      filters: filterTerms,
+      attributes
+    })
+      .catch(err => commonErrorHandler(err, `There was an error getting system devices device ${id}.`, dispatch, 'Please check your connection.'))
+      .then(({ data, headers }) => {
+        const state = getState();
+        const { devicesById, ids } = reduceReceivedDevices(data, [], state);
+        const device = {
+          ...state.devices.byId[id],
+          systemDeviceIds: ids,
+          systemDeviceTotal: Number(headers[headerNames.total])
+        };
+        return Promise.resolve(
+          dispatch({
+            type: DeviceConstants.RECEIVE_DEVICES,
+            devicesById: {
+              ...devicesById,
+              [id]: device
+            }
+          })
+        );
+      });
+  };
+
+export const getGatewayDevices = deviceId => (dispatch, getState) => {
+  const state = getState();
+  let device = state.devices.byId[deviceId];
+  const { attributes = {} } = device;
+  const { mender_gateway_system_id } = attributes;
+  const filters = [
+    { ...emptyFilter, key: 'mender_is_gateway', value: true, scope: 'system' },
+    { ...emptyFilter, key: 'mender_gateway_system_id', value: mender_gateway_system_id, scope: 'system' }
```

Comment:
Scope is `inventory`, it cannot be `system`. At the same time, the `mender_is_gateway` value is a string, not a boolean; the reason is that inventory attributes only support strings.

```suggestion
    { ...emptyFilter, key: 'mender_is_gateway', value: 'true', scope: 'inventory' },
    { ...emptyFilter, key: 'mender_gateway_system_id', value: mender_gateway_system_id, scope: 'inventory' }
```